cmake/multi_image.cmake

```diff
@@ -125,6 +125,82 @@ function(add_child_image_from_source)
   set(${ACI_NAME}_BOARD ${ACI_BOARD})
 
   message("\n=== child image ${ACI_NAME} - ${ACI_DOMAIN}${inherited} begin ===")
+
+  # Check if the folder 'child_image/<ACI_NAME>' exists, and apply all
+  # relevant kconfig sources.
+  #
+  # <sample> DIRECTORY
+  # | - prj.conf (A)
+  # | - prj_<desc>.conf (B)
+  # | - boards DIRECTORY
+  # | | - <board>.conf (C)
+  # | | - <board>_<desc>.conf (D)
+  # | - child_image DIRECTORY
+  # | | <ACI_NAME> DIRECTORY
+  # | | - boards DIRECTORY
+  # | | | - <board>.conf (E)        If present, use instead of (C)
+  # | | | - <board>_<desc>.conf (F) If present, use instead of (D)
+  # | | - prj.conf (G)              If present, use instead of (A)
+  # | |                             Note that (C) is ignored if this is present.
+  # | |                             Use (E) instead.
+  # | | - prj_<desc>.conf (H)       If present, used instead of (B) when user
+  # | |                             specify `-DCONF_FILE=prj_<desc>.conf for
+  # | |                             parent image. Note that any (C) is ignored
+  # | |                             if this is present. Use (F) instead.
+  # | | - <ACI_name>_prj.conf (I)   Fragment, used together with (A) and (C)
+  # | | - <ACI_name>_prj_<desc>.conf (J) Fragment, used together with (B) and (D)
+
+  set(alt_conf ${APPLICATION_SOURCE_DIR}/child_image/${ACI_NAME})
+  if (EXISTS ${alt_conf})
```

Comment:
I assume that if user specifies `-D${ACI_NAME}_CONF_FILE=myfile.conf` then that should overrule everything done here ?

Reply:
Correct, so we need to check if ${ACI_NAME}_CONFI_FILE is set, if it is, then we skip all of this.

Reply:
done